From 5287cd3cc3351d4787f52a12372eb23fd53ef63f Mon Sep 17 00:00:00 2001
From: Alexey Minnekhanov <alexeymin@postmarketos.org>
Date: Wed, 6 Dec 2023 13:51:34 +0300
Subject: [PATCH] iommu/arm-smmu-qcom: Add SDM630/660 mdp5 compatibles
 for identity

Add SDM630/660's MDP5 compatibles to switch the default iommu domain
type to IDENTITY, as similarly required by SDM845 and others.

Signed-off-by: Alexey Minnekhanov <alexeymin@postmarketos.org>

Source-Commit: https://github.com/sdm660-mainline/linux/commit/5287cd3cc3351d4787f52a12372eb23fd53ef63f
Scope: SDM630/660-common
Porting notes: Original MDP5 identity-domain match applies after the preceding SDM660 MSS patch.
---
diff --git a/drivers/iommu/arm/arm-smmu/arm-smmu-qcom.c b/drivers/iommu/arm/arm-smmu/arm-smmu-qcom.c
index 02531bc799629f..3b3a700bbfdaab 100644
--- a/drivers/iommu/arm/arm-smmu/arm-smmu-qcom.c
+++ b/drivers/iommu/arm/arm-smmu/arm-smmu-qcom.c
@@ -420,6 +420,8 @@ static const struct of_device_id qcom_smmu_client_of_match[] __maybe_unused = {
 	{ .compatible = "qcom,sc7280-mss-pil" },
 	{ .compatible = "qcom,sc8180x-mdss" },
 	{ .compatible = "qcom,sc8280xp-mdss" },
+	{ .compatible = "qcom,sdm630-mdp5" },
+	{ .compatible = "qcom,sdm660-mdp5" },
 	{ .compatible = "qcom,sdm660-mss-pil" },
 	{ .compatible = "qcom,sdm670-mdss" },
 	{ .compatible = "qcom,sdm845-mdss" },

